src/hocon.erl

```diff
@@ -400,11 +400,29 @@ do_concat(Concat) ->
 do_concat([], [{}]) ->
     {[]};
 do_concat([], [Object| _Objects] = Acc) when is_tuple(Object) ->
-    {lists:reverse(Acc)};
+    case lists:all(fun is_tuple/1, Acc) of
+        true ->
+            {lists:reverse(Acc)};
+        false ->
+            % TODO: print line number
```

Comment:
line number info has been dropped at this point, right ?

Reply:
No, line number info is discarded in parser, when we use `value_of(Token) -> element(3, Token).`.

Reply:
right, in that case, we can raise the error with the line number for the first element ?
i.e. no to print it.

Reply:
We can do it after we make the parser preserve line number.